test/Interpreter/anyhashable_casts.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %target-build-swift %s -o %t/main
+// RUN: %target-build-swift %s -O -o %t/main-optimized
+// RUN: %target-run %t/main | FileCheck %s
+// RUN: %target-run %t/main-optimized | FileCheck %s
+// REQUIRES: executable_test
+
+print("Begin")
+// CHECK: Begin
+
+let i = 5
+print(i)
+// CHECK-NEXT: 5
+
+let i_ah: AnyHashable = i
+print(i_ah)
+// CHECK-NEXT: 5
+
+let i_ah_i = i_ah as? Int
+print(i_ah_i)
+// CHECK-NEXT: Optional(5)
+
+let i_any: Any = i
+let i_any_ah = i_any as? AnyHashable
+print(i_any_ah)
+// CHECK-NEXT: Optional(AnyHashable(5))
+
+print("End")
+// CHECK-NEXT: End
```

Comment:
It would be good to do something like the logic in `test/1_stdlib/BridgeIdAsAny.swift.gyb`, to test all the possible cases of compounded existential/AnyHashable erasure:

https://github.com/apple/swift/blob/master/test/1_stdlib/BridgeIdAsAny.swift.gyb#L177